drivers/volume/gcp/gcp.go

```diff
@@ -0,0 +1,356 @@
+		}
+		volumeInfo.BackupID = snapshot.Name
+		volumeInfo.Options = map[string]string{
+			"projectID": g.projectID,
+		}
+	}
+	return volumeInfos, nil
+}
+
+func (g *gcp) GetBackupStatus(backup *storkapi.ApplicationBackup) ([]*storkapi.ApplicationBackupVolumeInfo, error) {
+	volumeInfos := make([]*storkapi.ApplicationBackupVolumeInfo, 0)
+
+	for _, vInfo := range backup.Status.Volumes {
+		if vInfo.DriverName != driverName {
+			continue
+		}
+		snapshot, err := g.service.Snapshots.Get(g.projectID, vInfo.BackupID).Do()
+		if err != nil {
+			return nil, err
+		}
+		switch snapshot.Status {
+		case "CREATING", "UPLOADING":
+			vInfo.Status = storkapi.ApplicationBackupStatusInProgress
+			vInfo.Reason = fmt.Sprintf("Volume backup in progress: %v", snapshot.Status)
+		case "DELETING", "FAILED":
+			vInfo.Status = storkapi.ApplicationBackupStatusFailed
+			vInfo.Reason = fmt.Sprintf("Backup failed for volume: %v", snapshot.Status)
+		case "READY":
+			vInfo.Status = storkapi.ApplicationBackupStatusSuccessful
+			vInfo.Reason = "Backup successful for volume"
+		}
+		volumeInfos = append(volumeInfos, vInfo)
+	}
+
+	return volumeInfos, nil
+
+}
+
+func (g *gcp) CancelBackup(backup *storkapi.ApplicationBackup) error {
+	return g.DeleteBackup(backup)
+}
+
+func (g *gcp) DeleteBackup(backup *storkapi.ApplicationBackup) error {
+	for _, vInfo := range backup.Status.Volumes {
+		if vInfo.DriverName != driverName {
+			continue
+		}
+		_, err := g.service.Snapshots.Delete(vInfo.Options["projectID"], vInfo.BackupID).Do()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *gcp) UpdateMigratedPersistentVolumeSpec(
+	pv *v1.PersistentVolume,
+) (*v1.PersistentVolume, error) {
+	if pv.Spec.CSI != nil {
+		pv.Spec.CSI.VolumeHandle = pv.Name
+		return pv, nil
+	}
+
+	pv.Spec.GCEPersistentDisk.PDName = pv.Name
+	return pv, nil
+}
+
+func (g *gcp) generatePVName() string {
+	return pvNamePrefix + string(uuid.NewUUID())
+}
+
+func (g *gcp) getSnapshotResourceName(
+	backupVolumeInfo *storkapi.ApplicationBackupVolumeInfo,
+) string {
+	return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%v/global/snapshots/%v",
+		backupVolumeInfo.Options["projectID"], backupVolumeInfo.BackupID)
+}
+
+func (g *gcp) StartRestore(
+	restore *storkapi.ApplicationRestore,
+	volumeBackupInfos []*storkapi.ApplicationBackupVolumeInfo,
+) ([]*storkapi.ApplicationRestoreVolumeInfo, error) {
+
+	volumeInfos := make([]*storkapi.ApplicationRestoreVolumeInfo, 0)
+	for _, backupVolumeInfo := range volumeBackupInfos {
+		volumeInfo := &storkapi.ApplicationRestoreVolumeInfo{}
+		volumeInfo.PersistentVolumeClaim = backupVolumeInfo.PersistentVolumeClaim
+		volumeInfo.SourceNamespace = backupVolumeInfo.Namespace
+		volumeInfo.SourceVolume = backupVolumeInfo.Volume
+		volumeInfo.RestoreVolume = g.generatePVName()
+		volumeInfo.DriverName = driverName
+		volumeInfos = append(volumeInfos, volumeInfo)
```

Comment:
NIT: could do the assignment in the append to have it be a one-liner.

Reply:
Need to use `volumeInfo` below, so can't do that. Moved the assignments inside the init of the object